src/par_iter/skip.rs

```diff
@@ -0,0 +1,78 @@
+use super::*;
+use super::internal::*;
+use std::cmp::min;
+
+pub struct Skip<M> {
+    base: M,
+    n: usize,
+}
+
+impl<M> Skip<M> 
+    where M: IndexedParallelIterator
+{
+    pub fn new(mut base: M, n: usize) -> Skip<M> {
+        let n = min(base.len(), n);
+        Skip { base: base,
+               n: n }
+    }
+}
+
+impl<M> ParallelIterator for Skip<M>
+    where M: IndexedParallelIterator
+{
+    type Item = M::Item;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge(self, consumer)
+    }
+}
+
+impl<M> ExactParallelIterator for Skip<M>
+    where M: IndexedParallelIterator
+{
+    fn len(&mut self) -> usize {
+        self.base.len() - self.n
+    }
+}
+
+impl<M> BoundedParallelIterator for Skip<M>
+    where M: IndexedParallelIterator
+{
+    fn upper_bound(&mut self) -> usize {
+        self.len()
+    }
+
+    fn drive<C: Consumer<Self::Item>>(self, consumer: C) -> C::Result {
+        bridge(self, consumer)
+    }
+}
+
+impl<M> IndexedParallelIterator for Skip<M>
+    where M: IndexedParallelIterator
+{
+    fn with_producer<CB>(self, callback: CB) -> CB::Output
+        where CB: ProducerCallback<Self::Item>
+    {
+        return self.base.with_producer(Callback { callback: callback,
+                                                  n: self.n});
+
+        struct Callback<CB> {
+            callback: CB,
+            n: usize,
+        }
+
+        impl<ITEM, CB> ProducerCallback<ITEM> for Callback<CB>
+            where CB: ProducerCallback<ITEM>
+        {
+            type Output = CB::Output;
+            fn callback<P>(self, base: P) -> CB::Output
+                where P: Producer<Item=ITEM>
+            {
+                let (_, producer) = base.split_at(self.n);
```

Comment:
So I guess if we were going to match the semantics of sequential `skip` we might change this to:

```rust
let (before_skip, after_skip) = base.split_at(self.n);
bridge_producer_consumer(before_skip, NoopConsumer::new());
self.callback.callback(after_skip);
```

(We'd have to write the no-op consumer, I don't think that exists.)

It feels a *bit* weird to call that *here*, in the `with_producer` method -- but it's probably the right place to do it. The other place would be `drive()` but, we'd still have to convert into a producer, and after all, all that `drive()` does is call `bridge()` which calls `with_producer()`.

We should also make some tests that observe side-effects from the skipped entries, e.g. using `map()` or `inspect()`.